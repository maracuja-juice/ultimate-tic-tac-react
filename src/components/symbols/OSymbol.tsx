import * as React from 'react';
import './symbols.css';
import './symbolAnimations.css';
import '../../fonts/icons.css';
import * as classNames from 'classnames';
import { CSSProperties } from 'react';

interface OSymbolProps {
    bigSymbol?: boolean;
    shouldAnimate?: boolean;
    style?: CSSProperties;
    cssClasses?: string;
}

export class OSymbol extends React.Component<OSymbolProps> {
    public static defaultProps: Partial<OSymbolProps> = {
        shouldAnimate: true
    };

    constructor( props: OSymbolProps ) {
        super( props );
    }

    render() {
        const {bigSymbol, shouldAnimate, style, cssClasses} = this.props;
        const iconClass = classNames( {
                                          'icon-o': true,
                                          'o': true,
                                          'big-symbol': bigSymbol,
                                          'animate-o': shouldAnimate
                                      } );

        return (
            <i style={style} className={iconClass + ' ' + cssClasses} aria-label="Circle Symbol"/>
        );
    }
}