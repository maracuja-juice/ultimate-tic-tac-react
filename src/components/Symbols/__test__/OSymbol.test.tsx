import * as React from 'react';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { OSymbol } from '../OSymbol';

configure({ adapter: new ReactSixteenAdapter() });

describe('OSymbol', () => {
  it('should not explode', () => {
    const component = shallow(<OSymbol />);
    expect(component).not.toBeNull();
  });

  it('should match snapshot', () => {
    const oSymbol1 = shallow(<OSymbol />);
    expect(oSymbol1).toMatchSnapshot();

    const oSymbol2 = shallow(<OSymbol shouldAnimate={false} />);
    expect(oSymbol2).toMatchSnapshot();

    const oSymbol3 = shallow(<OSymbol bigSymbol={true} />);
    expect(oSymbol3).toMatchSnapshot();
  });

  it('should have i element with some kind of icon', () => {
    const component = shallow(<OSymbol />);
    expect(component.find('i')).toHaveLength(1);
    expect(component.find('i').text()).not.toBeUndefined();
  });

  it('should have icon-o, o and animate-o class by default', () => {
    const component = shallow(<OSymbol />);
    expect(component.find('i').hasClass('icon-o')).toBe(true);
    expect(component.find('i').hasClass('o')).toBe(true);
    expect(component.find('i').hasClass('animate-o')).toBe(true);
    expect(component.find('i').hasClass('big-symbol')).toBe(false);
  });

  it('should have big-symbol and animate-o class with bigSymbol=true', () => {
    const component = shallow(<OSymbol bigSymbol={true} />);
    expect(component.find('i').hasClass('big-symbol')).toBe(true);
    expect(component.find('i').hasClass('animate-o')).toBe(true);
  });

  it(`shouldn't have animate class with shouldAnimate=false`, () => {
    let component = shallow(<OSymbol bigSymbol={true} shouldAnimate={false} />);
    expect(component.find('i').hasClass('animate-o')).toBe(false);

    component = shallow(<OSymbol shouldAnimate={false} />);
    expect(component.find('i').hasClass('animate-o')).toBe(false);
  });
});
