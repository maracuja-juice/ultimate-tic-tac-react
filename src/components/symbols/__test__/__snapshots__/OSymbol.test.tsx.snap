// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSymbol should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OSymbol
    shouldAnimate={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Circle Symbol",
      "className": "icon-o o animate-o undefined",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Circle Symbol",
        "className": "icon-o o animate-o undefined",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`OSymbol should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OSymbol
    shouldAnimate={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Circle Symbol",
      "className": "icon-o o undefined",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Circle Symbol",
        "className": "icon-o o undefined",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`OSymbol should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OSymbol
    bigSymbol={true}
    shouldAnimate={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Circle Symbol",
      "className": "icon-o o big-symbol animate-o undefined",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Circle Symbol",
        "className": "icon-o o big-symbol animate-o undefined",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
